nessus-agent/elx/configure.sls

```diff
@@ -0,0 +1,61 @@
+# This salt state configures a previously-installed Nessus Agent -
+# whether baked-in or installed by the 'install' component of this
+# formula. The agent will be configured and linked to the Nessus server
+# using the provided parameters (Nessus 64 hexadecimal-digit key,
+# Nessus server FQDN and port, Nessus Agent Group Name). The final step
+# is to start the Nessus Agent service.
+#
+###########################################################################
+{%- from tpldir ~ '/map.jinja' import nessus with context %}
+{%- set chkFile = '/etc/tenable_tag' %}
+{%- set staleDbs = salt.file.find('/opt/nessus_agent/var/nessus/', maxdepth='1', type='f', name='*.db') %}
+{%- set linkStr =
+        nessus.sbin_file ~
+        ' agent link --key=' ~ nessus.nessus_key ~
+        ' --host=' ~ nessus.nessus_server ~
+        ' --port=' ~ nessus.nessus_port ~
+        ' --groups=' ~ nessus.nessus_groups
+%}
+
+{%- if salt.file.file_exists(chkFile) %}
+Unlink Stale Agent-config:
+  cmd.run:
+    - name: '{{ nessus.sbin_file }} agent unlink'
+    - success_retcodes:
+      - 0
+      - 2
+    - unless: {{ linkStr }}
+
+Stop Nessus Agent:
+  service.dead:
+    - name: {{ nessus.package | lower }}
+    - require:
+      - cmd: Unlink Stale Agent-config
+
+  {%- for staleDb in staleDbs %}
+Nuke Stale {{ staleDb }} file:
+  file.absent:
+    - name: '{{ staleDb }}'
+    - require:
+      - Stop Nessus Agent
+    - require_in:
+      - Configure Nessus Agent
+  {%- endfor %}
+{%- endif %}
+
+Handle {{ chkFile }} file:
+  file.absent:
+    - name: '{{ chkFile }}'
+
+Configure Nessus Agent:
```

Comment:
Previously, this state was dependent on the link parameters being available.  Instead of an error, we would just print a notification.  This would allow the install to continue without error.  The agent just wouldn't be linked.

Reply:
Is this comment just informational (i.e., can we hit "resolve conversation" on your comment)?

Reply:
It depends.  Does the error break the Watchmaker execution?

Reply:
We want to be able to continue even if the link is not successful.

Reply:
The idea is that we shouldn't execute this `configure.sls` in environments where it isn't supported. We may have to break this into phases to get there. I.e.

* Leave init.sls alone, duplicating the functionality
* Add install.sls and configure.sls
* Release watchmaker with this in place
* Modify salt-content to call *only* the install.sls
* Update this formula to modify init.sls
* Release watchmaker again

Reply:
Install runs first, so, it _always_ happens (assuming the pillar contains reference to a valid RPM URL) ...but, if the pillar is incomplete (arguably, if you're doing a production installation of the Nessus agent, the associated pillar should never be incomplete – the failure I'm hitting in DICELAB appears to be tied to incomplete pillar-data). That said, to support the DICELAB-type of testing-condition, it may warrant putting the pillar-check back.

Reply:
> We want to be able to continue even if the link is not successful.

My assumption would have been that one would want the provisioning-process to fail-out if a required enterprise integration breaks?